plutus-core/plutus-ir/src/PlutusIR/Compiler/Definitions.hs

```diff
@@ -88,9 +88,9 @@ runDefT x act = do
         where
             bindingDefs defs =
                 let
-                    terms = mapDefs (\d -> TermBind x (snd $ PLC.defVal d) (PLC.defVar d) (fst $ PLC.defVal d)) (_termDefs defs)
-                    types = mapDefs (\d -> TypeBind x (PLC.defVar d) (PLC.defVal d)) (_typeDefs defs)
-                    datatypes = mapDefs (\d -> DatatypeBind x (PLC.defVal d)) (_datatypeDefs defs)
+                    terms = mapDefs (\d -> TermBind (_varDeclAnn (defVar d)) (snd $ PLC.defVal d) (PLC.defVar d) (fst $ PLC.defVal d)) (_termDefs defs)
```

Comment:
Worth a comment, maybe? We talked about this a fair bit, so it would be good to record it. Or maybe we even need a Note that we can refer to from the docs of the other functions, since this is something you might need to know about in order to use e.g. `defineTerm`.

Reply:
Yeah, I do think we need a Note to explain why we are taking annotation from var, not val. I'll open a separate PR for it since it will be a non-trivial note.

Reply:
If it's not blocking you I'd generally rather do documentation in the same PR as the code.

Reply:
Sure, I'm ok that .